spark/src/main/scala/org/apache/spark/sql/delta/commands/DeleteCommand.scala

```diff
@@ -509,5 +529,8 @@ case class DeleteMetric(
     numBytesRemoved: Long,
     changeFileBytes: Long,
     scanTimeMs: Long,
-    rewriteTimeMs: Long
+    rewriteTimeMs: Long,
+    numDeletionVectorsAdded: Long,
+    numDeletionVectorsRemoved: Long,
+    numDeletionVectorsUpdated: Long
```

Comment:
```suggestion
    numDeletionVectorsAdded: Long,
    numDeletionVectorsRemoved: Long,
    numDeletionVectorsUpdated: Long
```

IMO these should be `Option[Long]` instead of `Long`, because we should not be emitting DV metrics (and enlarge log size) when DV's not enabled.

Reply:
@larsk-db whats your opinion on this ?

Reply:
Sorry for the confusion -- I mean these metrics should only be in the JSON when actually being set to some numbers. For example, when "we have DV files but DV are not enabled", `numDeletionVectorsAdded` should be `None` instead of `0`.

Reply:
sound ok to me let me make the change will revert if big bosses not like it .

Reply:
@xupefei When a SQL metric is created, it is initialized with a default value of 0. If the metric is not created, it cannot be updated. If the metric is created and left unchanged, it will populate with a value of 0.
any idea how i can avoid this pit ?
  https://github.com/delta-io/delta/blob/3bd11de9b42731074768abeeb0cd984d14d99b85/spark/src/main/scala/org/apache/spark/sql/delta/commands/DeleteCommand.scala#L67-L69

Reply:
Imo SQL metric is fine to always have these three metrics defined, as these metrics won't be written to the Delta log.
Our goal is to make `DeleteMetric` (which is serialized to Delta log) not have a DV metric when not needed.